frontend/src/app/data-entry/shared/episode-link-form/episode-link-form.component.ts

```diff
@@ -0,0 +1,174 @@
+import { FormControl, FormGroup } from '@angular/forms';
+import { entityTypeNames, formStatusSubject, sourceMentionSelectOptions } from '../utils';
+import _ from 'underscore';
+import { MutationResult } from 'apollo-angular';
+import { FormService } from '../form.service';
+
+
+type LinkTo = 'episode' | 'entity';
+let nextID = 0;
+
+@Component({
+    selector: 'lc-episode-link-form',
+    templateUrl: './episode-link-form.component.html',
+    styleUrls: ['./episode-link-form.component.scss']
+})
+export class EpisodeLinkFormComponent implements OnChanges, OnDestroy {
+    /** ID of the connection object (e.g. EpisodeAgent) */
+    @Input({ required: true }) entityID!: string;
+    @Input({ required: true }) episodeID!: string;
+
+    /** Type of entity (agent, letter, etc.) */
+    @Input({ required: true }) entityType!: Entity;
+
+    /** Describes the direction of the link in the document.
+     *
+     * The header will show the model that is being linked *to*; the model being linked
+     * *from* should be clear from the context.
+     */
+    @Input() linkTo: LinkTo = 'episode';
+
+
+    data$: Observable<DataEntryEpisodeEntityLinkQuery | undefined>;
+    form = new FormGroup({
+        sourceMention: new FormControl<SourceMention>(SourceMention.Direct),
+        note: new FormControl<string>('', { nonNullable: true }),
+    });
+
+    sourceMentionOptions: SelectOptions<SourceMention> = sourceMentionSelectOptions();
+
+    private link$ = new Subject<UpdateEpisodeEntityLinkInput>();
+    private status$ = formStatusSubject();
+    private id = `episode-link-${nextID++}`;
+
+    actionIcons = actionIcons;
+
+    constructor(
+        query: DataEntryEpisodeEntityLinkGQL,
+        private updateMutation: DataEntryUpdateEpisodeEntityLinkGQL,
+        private formService: FormService,
+    ) {
+        this.formService.attachForm(this.id, this.status$);
+
+        this.data$ = this.link$.pipe(
+            switchMap((link) => query.watch(link).valueChanges),
+            map(result => result.data),
+            shareReplay(1),
+            takeUntilDestroyed(),
+        );
+        this.data$.subscribe(this.updateFormValues.bind(this));
+
+        this.form.valueChanges.pipe(
+            debounceTime(500),
+            distinctUntilChanged(_.isEqual),
+            skip(1),
+            tap(() => this.status$.next('loading')),
+            withLatestFrom(this.link$),
+            map(this.toMutationInput),
+            switchMap(this.makeMutation.bind(this)),
+            takeUntilDestroyed(),
+        ).subscribe(this.onMutationResult.bind(this));
+    }
+
+    get sourceMentionInputID() {
+        return this.id + '-source-mention-input';
+    }
+
+    get notesInputID() {
+        return this.id + '-notes-input';
+    }
+
+    get entityName(): string {
+        return entityTypeNames[this.entityType];
+    }
+
+    get linkedObjectName(): string {
+        return this.linkTo == 'episode' ? 'episode' : this.entityName;
```

Comment:
```suggestion
        return this.linkTo == 'episode' ?? this.entityName;
```

(All of these getters would be sweet `computed()`s by the way, but that's something to consider when we're less strapped for time.)